plugins/frontity-headtags/includes/class-frontity-headtags.php

```diff
@@ -0,0 +1,262 @@
+<?php
+/**
+ * File class-frontity-headtags.php
+ *
+ * @package Frontity_Headtags
+ */
+
+/**
+ * Class that contains the main behavior.
+ *
+ * This class has methods to get head tags by entity (posts, pages, taxonomies, authors and archives),
+ *
+ * These methods deal with three tasks mainly:
+ * - add head tags to cache and invalidate it (using transients)
+ * - compute and parse head tags by entity
+ * - replace wp_query in order to compute head tags
+ */
+class Frontity_Headtags {
+	/**
+	 * Variable to store the original WP Query object (needed to restore it).
+	 *
+	 * @access  private
+	 * @var     array
+	 */
+	private $query_backup = array();
+
+	/**
+	 * Get cached head tags if found.
+	 *
+	 * @param string $key Transient key.
+	 * @return array|mixed
+	 */
+	public function get_cached_headtags( $key ) {
+		return get_transient( "frontity_headtags_{$key}" );
+	}
+
+	/**
+	 * Set cached head tags for this key.
+	 *
+	 * @param string      $key Transient key.
+	 * @param array|mixed $headtags Head tags to cache.
+	 * @return bool
+	 */
+	public function set_cached_headtags( $key, $headtags ) {
+		$transient_key = "frontity_headtags_{$key}";
+
+		// Get the list of transients.
+		$transient_list = get_option( 'frontity_headtags_transients', array() );
+
+		if ( ! in_array( $transient_key, $transient_list, true ) ) {
+
+			// Add the transient key if is not in the list.
+			$transient_list[] = $transient_key;
+			set_option( 'frontity_headtags_transients', $transient_list );
```

Comment:
My WordPress complains that `set_option` doesn't exist. I searched in Google but it seems like there's no such function. I've replaced it for `update_option` in my local environment and it looks like it works fine.

Where did you get `set_option`from?

Reply:
Changed in https://github.com/frontity/wp-plugins/pull/10/commits/9eeea8cf6970362249f422a8e320ebbfbd405c8f.